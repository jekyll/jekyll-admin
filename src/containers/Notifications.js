import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import NotificationSystem from 'react-notification-system';

export class Notifications extends Component {
  componentDidMount() {
    this.notificationSystem = this.refs.notificationSystem;
  }

  componentWillReceiveProps(nextProps) {
    const { notification } = nextProps;
    this.notificationSystem.addNotification({
      title: notification.title,
      message: notification.message,
      level: notification.level,
      position: 'br',
      autoDismiss: 10,
    });
  }

  render() {
    return <NotificationSystem ref="notificationSystem" />;
  }
}

Notifications.propTypes = {
  notification: PropTypes.object.isRequired,
};

const mapStateToProps = state => ({
  notification: state.notifications.notification,
});

export default connect(mapStateToProps)(Notifications);
