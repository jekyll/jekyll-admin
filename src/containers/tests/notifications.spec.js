import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import { mount } from 'enzyme';

import { Notifications } from '../Notifications';
import { notification } from './fixtures';

describe('Containers::Notification', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<Notifications notification={notification} />, div);
  });

  it('renders correctly', () => {
    const tree = renderer
      .create(<Notifications notification={notification} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders correctly after props change', () => {
    const notif = mount(<Notifications notification={notification} />);
    const newNotification = {
      title: 'Test Again',
      message: 'Testing notifications change',
      level: 'success',
    };
    notif.setProps({ notification: newNotification });
    expect(notif.find('.notification-message').text()).toBe(
      'Testing notifications change'
    );
  });
});
