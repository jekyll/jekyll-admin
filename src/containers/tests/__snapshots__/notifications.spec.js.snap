// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Containers::Notification renders correctly 1`] = `
<div
  className="notifications-wrapper"
  style={Object {}}
/>
`;
